clang/lib/AST/ComputeDependence.cpp

```diff
@@ -376,6 +376,9 @@ ExprDependence clang::computeDependence(PackExpansionExpr *E) {
 
 ExprDependence clang::computeDependence(PackIndexingExpr *E) {
   ExprDependence D = E->getIndexExpr()->getDependence();
+  if (D & ExprDependence::Value)
+    D |= ExprDependence::TypeInstantiation;
```

Comment:
Does it mean "if the index expression is value-dependent, then the whole `PackIndexingExpr` is type-dependent despite its expressions"? For example,
```cpp
template <int N, int... E>
auto foo() {
  return E...[N];
}
```
Should `E...[N]` be treated as type-dependent regardless of the result being always an int?

Reply:
Probably `E->getPackIdExpression()->getDependence() & ExprDependence::TypeInstantiation`?

Reply:
> Should E...[N] be treated as type-dependent regardless of the result being always an int?

Yes. (We don't do anything to look at whether all the expressions would have the same type)

Reply:
NVM, you are right, we can simplify this whole thing, I'll do that shortly